cmd/registry/core/lint-openapi.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/apigee/registry/rpc"
+	linter "github.com/googleapis/gnostic/metrics/lint"
+	"google.golang.org/protobuf/proto"
+	yaml "gopkg.in/yaml.v3"
+)
+
+// NewLintFromOpenAPI runs the API linter and returns the results.
+func NewLintFromOpenAPI(name string, b []byte) (*rpc.Lint, error) {
+	// create a tmp directory
+	root, err := ioutil.TempDir("", "registry-openapi-")
```

Comment:
Could anything go wrong if this directory isn't cleaned up properly? Using `ioutil.TempDir("", "registry-openapi-*")` would result in different directories between runs.

Reply:
I may have misunderstood the docs. My impression was that the directory name would only have randomness if the asterisk is explicitly provided. If a random string is added to the end when no asterisk is provided this should be fine.